src/celrender/atmosphererenderer.cpp

```diff
@@ -0,0 +1,396 @@
+                else if (cosSunAngle >= 0.3f)
+                    brightness = 1.0f;
+                else
+                    brightness = (cosSunAngle + 0.2f) * 2.0f;
+            }
+
+            memcpy(&vtx.position[0], p.data(), vtx.position.size() * sizeof(vtx.position[0]));
+
+            float atten = 1.0f - hh;
+            Eigen::Vector3f color = vecgl::mix(botColor, topColor, hh);
+            brightness *= minOpacity + (1.0f - minOpacity) * fade * atten;
+            if (coloration != 0.0f)
+                color = vecgl::mix(color, sunsetColor, coloration);
+
+            Color(brightness * color.x(),
+                  brightness * color.y(),
+                  brightness * color.z(),
+                  fade * (minOpacity + (1.0f - minOpacity)) * atten).get(&vtx.color[0]);
+            m_skyVertices.push_back(vtx);
+        }
+    }
+    m_skyContour.clear();
+
+    // Create the index list
+    BuildIndexList(static_cast<ushort>(nRings), static_cast<ushort>(nSlices), m_skyIndices);
+}
+
+void
+AtmosphereRenderer::initVertexObject()
+{
+    m_vo->bind();
+    m_vo->setVertexAttribArray(
+        CelestiaGLProgram::VertexCoordAttributeIndex,
+        3,
+        GL_FLOAT,
+        false,
+        sizeof(SkyVertex),
+        offsetof(SkyVertex, position));
+    m_vo->setVertexAttribArray(
+        CelestiaGLProgram::ColorAttributeIndex,
+        4,
+        GL_UNSIGNED_BYTE,
+        true,
+        sizeof(SkyVertex),
+        offsetof(SkyVertex, color));
+}
+
+void
+AtmosphereRenderer::renderLegacy(
+    const Atmosphere         &atmosphere,
+    const LightingState      &ls,
+    const Eigen::Vector3f    &center,
+    const Eigen::Quaternionf &orientation,
+    const Eigen::Vector3f    &semiAxes,
+    const Eigen::Vector3f    &sunDirection,
+    float                     pixSize,
+    bool                      lit,
+    const Matrices           &m)
+{
+    computeLegacy(atmosphere, ls, center, orientation, semiAxes, sunDirection, pixSize, lit);
+
+    ShaderProperties shadprop;
+    shadprop.texUsage = ShaderProperties::VertexColors;
+    shadprop.lightModel = ShaderProperties::UnlitModel;
+    auto *prog = m_renderer.getShaderManager().getShader(shadprop);
+    if (prog == nullptr)
+        return;
+
+    Renderer::PipelineState ps;
+    ps.depthTest = true;
+    ps.blending = true;
+    ps.blendFunc = {GL_ONE, GL_ONE_MINUS_SRC_ALPHA};
+    m_renderer.setPipelineState(ps);
+
+    if (m_vo->initialized())
+        m_vo->bindWritable();
+    else
+        initVertexObject();
+
+    m_vo->allocate(nullptr, nullptr); // allocate or orphan GPU memory buffers
+
+    m_vo->setBufferData(m_skyVertices.data(), 0, m_skyVertices.size() * sizeof(SkyVertex));
+    m_vo->setIndexBufferData(m_skyIndices.data(), 0, m_skyIndices.size() * sizeof(unsigned short));
+
+    prog->use();
+    prog->setMVPMatrices(*m.projection, *m.modelview);
+    m_vo->draw(GL_TRIANGLE_STRIP, static_cast<int>(m_skyIndices.size()), 0);
```

Comment:
I am getting a segmentation fault here.

Reply:
what object?

Reply:
Crashes as soon as the Sun comes into view during the startup script.

Reply:
If I replace this method with just `return` (disabling legacy atmospheres), I can also test the comet tails. They also segfault.